### PR TITLE
Add basic GoogleAnalytics to to project

### PR DESCRIPTION
**Pull request by ryancarlson**
Initially we will track a handful of screens to make sure we get the basic number of users, sessions, rough location etc.

The mechanism is for each VC to override the func `screenName()` to provide its screen name to the analytics handler.

### CODE REVIEW

**Comment by ryancarlson** [comment]
@michaelharro would approve of this, I bet ;)

File: godtools/ViewControllers/BaseViewController.swift [hunk 1]
```diff
@@ -183,4 +184,18 @@ class BaseViewController: UIViewController {
...
+        let userInfo = [GTConstants.kAnalyticsScreenNameKey: screenName()]
+        
+        NotificationCenter.default.post(name: .screenViewNotification,
+                                        object: nil,
+                                        userInfo: userInfo)
+    }
+    func screenName() -> String {
+//        assertionFailure("method must be overriden by view controller")
+        
+        return "unknown"
```
Comment (pablomarti) [inline godtools/ViewControllers/BaseViewController.swift]: Everything looks good, just this, but will this be updated later?

Comment (ryancarlson) [inline godtools/ViewControllers/BaseViewController.swift]: Yes, I wanted to confer with @frett to make sure we have the same naming pattern for our screens on both platforms.

**Comment by ryancarlson** [comment]
@pablomarti is this okay to merge so that other PRs can move forward? I will create an issue to add other screen names to the analytics tracking.

**Pull request merged**